File: cast-extractor/src/main/java/com/libreshockwave/tools/format/PaletteDescriptions.java
package com.libreshockwave.tools.format;

/**
 * Utility for getting human-readable palette descriptions.
 */
public final class PaletteDescriptions {

    private PaletteDescriptions() {}

    /**
     * Returns a human-readable description for a palette ID.
     */
    public static String get(int paletteId) {
        return switch (paletteId) {
            case -1 -> "System Mac";
            case -2 -> "Rainbow";
            case -3 -> "Grayscale";
            case -4 -> "Pastels";
            case -5 -> "Vivid";
            case -6 -> "NTSC";
            case -7 -> "Metallic";
            case -101 -> "System Windows";
            case -102 -> "System Windows (D4)";
            default -> paletteId >= 0 ? "Cast Member #" + (paletteId + 1) : "Unknown (" + paletteId + ")";
        };
    }
}

File: cpp/src/editor/format/PaletteDescriptions.cpp
#include "libreshockwave/editor/format/PaletteDescriptions.hpp"

namespace libreshockwave::editor::format {

std::string PaletteDescriptions::get(int paletteId) {
    switch (paletteId) {
        case -1:
            return "System Mac";
        case -2:
            return "Rainbow";
        case -3:
            return "Grayscale";
        case -4:
            return "Pastels";
        case -5:
            return "Vivid";
        case -6:
            return "NTSC";
        case -7:
            return "Metallic";
        case -101:
            return "System Windows";
        case -102:
            return "System Windows (D4)";
        default:
            if (paletteId >= 0) {
                return "Cast Member #" + std::to_string(paletteId + 1);
            }
            return "Unknown (" + std::to_string(paletteId) + ")";
    }
}

} // namespace libreshockwave::editor::format
